packages/aws-rfdk/lib/deadline/lib/secrets-management.ts

```diff
@@ -0,0 +1,262 @@
+      throw new Error('Repository does not contain secrets management credentials');
+    }
+    this.adminCredentials = props.repository.secretsManagementSettings.credentials;
+    this.deploymentInstance = props.deploymentInstance;
+
+    // Download and install the Deadline Client
+    this.installDeadlineClient(props);
+
+    // Configure the Deadline Client to direct-connect to the repository
+    props.repository.configureClientInstance({
+      host: props.deploymentInstance,
+      mountPoint: '/mnt/repository',
+    });
+
+    // Install python dependencies
+    this.preparePythonEnvironment(props);
+    const localScriptFile = this.preparePythonScript(props);
+    this.runPythonScript(props, localScriptFile);
+
+    props.deploymentInstance.addExecutionDependency(props.repository);
+  }
+
+  public addSubnetIdentityRegistrationSetting(addSubnetProps: SubnetIdentityRegistrationSettingsProps) {
+    if (addSubnetProps.role === SecretsManagementRole.ADMINISTRATOR) {
+      throw new Error('The Administrator role cannot be set using a Deadline identity registration setting');
+    }
+    const { vpc, vpcSubnets } = addSubnetProps;
+    const selectedSubnets = vpc.selectSubnets(vpcSubnets);
+    selectedSubnets.subnets.forEach(subnet => {
+      const observedSubnet = this.subnetRegistrations.get(subnet.subnetId);
+      if (observedSubnet) {
+        if (observedSubnet.registrationStatus !== addSubnetProps.registrationStatus) {
+          throw new Error(`Subnet is already registered with registrationStatus "${observedSubnet.registrationStatus}" but another caller requested "${addSubnetProps.registrationStatus}"`);
+        } else if (observedSubnet.role !== addSubnetProps.role) {
+          throw new Error(`Subnet is already registered with role "${observedSubnet.role}" but another caller requested "${addSubnetProps.role}"`);
+        }
+      } else {
+        this.subnetRegistrations.set(subnet.subnetId, {
+          registrationStatus: addSubnetProps.registrationStatus,
+          role: addSubnetProps.role,
+          stack: Stack.of(subnet),
+        });
+      }
+    });
+    addSubnetProps.dependent.node.addDependency(this.deploymentInstance);
+  }
+
+  private runPythonScript(props: SecretsManagementIdentityRegistrationProps, localScriptFile: string) {
+    // The command-line arguments to be passed to the script that configures the Deadline identity registration
+    // settings
+    const scriptArgs = Lazy.list({
+      produce: () => {
+        return ([] as string[]).concat(
+          [
+            // Region
+            '--region',
+            Stack.of(this).region,
+            // Admin credentials
+            '--credentials',
+            `"${this.adminCredentials.secretArn}"`,
+          ],
+          // Subnets of the load balancer
+          (
+            props.renderQueueSubnets
+              .subnetIds
+              .map(subnetID => `--connection-subnet "${subnetID}"`)
+          ),
```

Comment:
This was done for horizontally aligning the commas in the outer list. What do you think? Does it help or hinder readability?

Reply:
I'm not sure what you mean by aligning the commas in the outer list. It might align it with the array above it, but it doesn't look like it lines up with the `Array.from()` item below it. Also I'm not sure if there should be a dangling comma on the line with the `.map()`.